packages/@aws-cdk/core/lib/bundling.ts

```diff
@@ -79,6 +79,40 @@ export interface BundlingOptions {
    * @experimental
    */
   readonly local?: ILocalBundling;
+
+  /**
+   * The type of packaging to apply on the bundling output.
+   *
+   * @default BundlePackaging.AUTO
+   *
+   * @experimental
+   */
+  readonly packaging?: BundlePackaging;
+}
+
+/**
+ * The type of packaging to apply on the bundled asset.
+ *
+ * @experimental
+ */
+export enum BundlePackaging {
+  /**
+   * The bundling output will always be zipped and uploaded to S3, regardless
+   * of its content.
+   */
+  ALWAYS_ZIP = 'zip',
+
+  /**
+   * The bundling output will not be zipped. Bundling will fail if the bundling
+   * output doesn't contain a single file.
```

Comment:
```suggestion
   * output doesn't contain a single archive .zip or .jar file.
```

Reply:
I chose to **allow** a single **non-archive** file for `NEVER_ZIP`, let's say you want to transform a PNG to JPG and upload it to S3... what do you think?

So `NEVER_ZIP` only throws when the bundling output is not a single file (archive or not).

Reply:
Not really a use case. If people want to publish files to S3 they should use s3-deployment. Assets should always be zip files.

Reply:
I'm ok with that but this means that without bundling you can use a single non archive file and with bundling you cannot.

Reply:
Yes, I am fine with that.